### PR TITLE
Make spec a required field in SBR

### PR DESCRIPTION
**Pull request by pratikjagrut**
### Motivation

Making spec a required field
https://issues.redhat.com/browse/APPSVC-711

### CODE REVIEW

File: test/acceptance/features/steps/steps.py [hunk 1]
```diff
@@ -394,3 +397,40 @@ def etc_cluster_is_running(context, etcd_name):
         print("etcd cluster not present, creating etcd cluster")
         assert etcd_cluster.create() is True, "etcd cluster is not created"
         assert etcd_cluster.is_present() is True, "etcd cluster is not present"
+
+
+@when(u'Invalid Service Binding is applied')
+def invalid_sbr_is_applied(context):
+    sbr = ServiceBinding()
+    # Get resource version of sbr if sbr is available
+    if context.__contains__("sbr_name"):
```
Comment (baijum) [inline test/acceptance/features/steps/steps.py:406]: I can see many uses of the special method `__contains__`. Is there any problem to use `in` operator?

Ref. https://stackoverflow.com/questions/1964934/what-does-contains-do-what-can-call-contains-function

Comment (pratikjagrut) [reply test/acceptance/features/steps/steps.py:406]: I didn't really think about it `__contains__` was already in use so I simply continued it but will look into it and if `in` works will replace `__contains__` with `in`.

Comment (pratikjagrut) [reply test/acceptance/features/steps/steps.py]: Hey updated as you suggested. PTAL.

File: test/acceptance/features/steps/steps.py [hunk 2]
```diff
@@ -394,3 +397,40 @@ def etc_cluster_is_running(context, etcd_name):
...
+def invalid_sbr_is_applied(context):
+    sbr = ServiceBinding()
+    # Get resource version of sbr if sbr is available
+    if "sbr_name" in context:
+        json_path = "{.metadata.resourceVersion}"
+        rv = sbr.get_servicebinding_info_by_jsonpath(context.sbr_name, context.namespace.name, json_path)
+        context.resource_version = rv
+    yaml = context.text
+    output = sbr.attempt_to_create(yaml)
+    context.expected_error = output
```
Comment (pedjak) [inline test/acceptance/features/steps/steps.py:412]: we can join these two lines:

```
context.expect_error = sbr.attemp_to_create(yaml)
```

File: test/acceptance/features/steps/steps.py [hunk 3]
```diff
@@ -394,3 +397,40 @@ def etc_cluster_is_running(context, etcd_name):
...
+
+
+@then(u'Error message "{err_msg}" is thrown')
+def validate_error(context, err_msg):
+    search = re.search(rf'.*{err_msg}.*', context.expected_error)
+    assert search is not None, "Actual error: {context.expected_error}, Expected error: {err_msg}"
+
+
+@then(u'Service Binding "{sb_name}" is not persistent in the cluster')
+def get_sb(context, sb_name):
```
Comment (pedjak) [inline test/acceptance/features/steps/steps.py:422]: can we update the function name to reflect what is going on? the function is not about getting  service binding - it is about checking if the given SB exists in the cluster.

File: test/acceptance/features/steps/steps.py [hunk 4]
```diff
@@ -394,3 +397,40 @@ def etc_cluster_is_running(context, etcd_name):
...
+    openshift = Openshift()
+    output = openshift.search_resource_in_namespace("servicebindings", sb_name, context.namespace.name)
+    assert output is None, "Service Binding {sb_name} is present in namespace {context.namespace.name}"
+
+
+@then(u'Service Binding "{sb_name}" is not updated')
+def get_persistent_sb(context, sb_name):
```
Comment (pedjak) [inline test/acceptance/features/steps/steps.py:429]: same as above.

**Review by pedjak — CHANGES_REQUESTED** [review]
see the comments.

File: test/acceptance/features/steps/steps.py (hunk 2 above)
Comment (pratikjagrut) [reply test/acceptance/features/steps/steps.py]: updated.

File: test/acceptance/features/steps/steps.py (hunk 3 above)
Comment (pratikjagrut) [reply test/acceptance/features/steps/steps.py]: Updated PTAL.

File: test/acceptance/features/steps/steps.py (hunk 4 above)
Comment (pratikjagrut) [reply test/acceptance/features/steps/steps.py]: Updated PTAL.

**Comment by pratikjagrut** [comment]
Forced push for squashing the commits.

**Pull request merged**